examples/llm/src/models/layers/flash_attention.py

```diff
@@ -58,14 +58,7 @@ def forward(
                 Binary and byte masks are supported.
```

Comment:
(Sorry this comment was supposed to be part of my review but it didn't take)

Is this a good time to axe `key_padding_mask` from this module? At the very least we should also update the text here because it implies that we're using this argument, but we're not because we're expecting `key_padding_mask` was already baked into `attn_mask`.

If you're good with axing stuff, `need_weights` and `average_attn_weights` are looking mighty deprecated too.

Reply:
I'll rm `need_weights` and `average_attn_weights` but keep `key_padding_mask` since it a very common arg. Keeping it enables me to throw an informative error.